test/gc.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+/* global WeakRef, FinalizationRegistry */
+
+const { test } = require('tap')
+const { Client } = require('..')
+const { createServer } = require('net')
+const FakeTimers = require('@sinonjs/fake-timers')
+
+const SKIP = typeof WeakRef === 'undefined' || typeof FinalizationRegistry === 'undefined'
+
+test('gc should collect the client if there are no active sockets', { skip: SKIP }, t => {
+  t.plan(4)
+
+  const clock = FakeTimers.install()
+  t.teardown(clock.uninstall.bind(clock))
+
+  const server = createServer((socket) => {
+    socket.write('HTTP/1.1 200 OK\r\n')
+    socket.write('Content-Length: 0\r\n')
+    socket.write('Keep-Alive: timeout=1s\r\n')
+    socket.write('Connection: keep-alive\r\n')
+    socket.write('\r\n\r\n')
+  })
+  t.teardown(server.close.bind(server))
+
+  let weakRef
+  let disconnected = false
+
+  const registry = new FinalizationRegistry((data) => {
+    console.log('Finalization')
+    t.strictEqual(data, 'test')
+    t.strictEqual(disconnected, true)
+    t.strictEqual(weakRef.deref(), undefined)
+  })
+
+  server.listen(0, () => {
+    const client = new Client(`http://localhost:${server.address().port}`, {
+      keepAliveTimeoutThreshold: 500
+    })
+    client.once('disconnect', () => {
+      disconnected = true
+      clock.tick(1e6)
```

Comment:
Why is this tick needed?

Reply:
This is the tick that allows for the `registry` callback to be called. Could it be that the fake clock interferes with the `gc` activity?

Reply:
Can you try without a fake clock?

Reply:
The test remains stuck waiting for the callback without the tick, even if I call the `gc` manually. Maybe I'm missing something.

Reply:
> Can you try without a fake clock?

Yes, let's try without the fake clock.

Reply:
Removed the fake clock and triggered `gc` manually. The timing is similar to before.